medallion/test/test_startup.py

```diff
@@ -0,0 +1,338 @@
+        with mock.patch(
+            "medallion.scripts.run.log"
+        ) as mock_logger, mock.patch(
+            "sys.argv", ["ARGV0", "--conf-check"]
+        ):
+            medallion.scripts.run.main()
+        mock_logger.setLevel.assert_called_once_with(logging.DEBUG)
+        mock_app.assert_not_called()
+    mock_app.reset_mock()
+
+    class ExpectedException(BaseException):
+        pass
+
+    with subtests.test(msg="--conf-check with equals"):
+        with mock.patch(
+            "sys.argv", ["ARGV0", "--conf-check=1"]
+        ), mock.patch(
+            "argparse.ArgumentParser.error", side_effect=ExpectedException,
+        ) as mock_error, pytest.raises(ExpectedException):
+            medallion.scripts.run.main()
+        mock_error.assert_called_once()
+        (expected_call, ) = mock_error.mock_calls
+        (msg, ) = expected_call[1]
+        assert "ignored explicit argument" in msg
+    mock_app.reset_mock()
+
+    with subtests.test(msg="--conf-check overrides --log-level"):
+        with mock.patch(
+            "medallion.scripts.run.log"
+        ) as mock_logger, mock.patch(
+            "sys.argv", ["ARGV0", "--conf-check", "--log-level=CRITICAL"]
+        ):
+            medallion.scripts.run.main()
+        mock_logger.setLevel.assert_called_once_with(logging.DEBUG)
+        mock_app.assert_not_called()
+
+
+def test_main_config_arg_handling(subtests):
+    """
+    Confirm that config arguments and options in `argv` are respected when run.
+    """
+    config_file_arg = "/we're/on/the/road/to/nowhere"
+    config_dir_arg = "/i/wanna/really/really/really/wanna/zigazig/dir/"
+    # We need to provide this to satisfy the backend module loading code
+    safe_config = {"backend": {"module_class": "MemoryBackend"}}
+    default_medallion_kwargs = {
+        "host": "127.0.0.1",
+        "port": 5000,
+        "debug": False,
+    }
+
+    with mock.patch(
+        "medallion.scripts.run.application_instance",
+    ) as mock_app, mock.patch(
+        "medallion.current_app", new=mock_app,
+    ), mock.patch(
+        "medallion.config.load_config", return_value=safe_config,
+    ) as mock_load_config:
+        with subtests.test(msg="No config args provided"):
+            with mock.patch(
+                "sys.argv", ["ARGV0"]
+            ):
+                medallion.scripts.run.main()
+            mock_load_config.assert_called_once_with(
+                medallion.config.DEFAULT_CONFFILE,
+                medallion.config.DEFAULT_CONFDIR,
+            )
+            mock_app.run.assert_called_once_with(**default_medallion_kwargs)
+        mock_app.reset_mock()
+        mock_load_config.reset_mock()
+
+        with subtests.test(msg="CONFIG_PATH provided only"):
+            with mock.patch(
+                "sys.argv", ["ARGV0", config_file_arg]
+            ):
+                medallion.scripts.run.main()
+            mock_load_config.assert_called_once_with(
```

Comment:
Fails with Windows system variables set. It expects `medallion.config.DEFAULT_CONFDIR` on the second call, but receives my environment variable instead.